src/rust/rust-proto/src/plugin_work_queue.rs

```diff
@@ -177,44 +249,52 @@ impl From<GetExecuteGeneratorRequest> for GetExecuteGeneratorRequestProto {
 }
 
 pub struct GetExecuteGeneratorResponse {
-    execution_job: ExecutionJob,
-    request_id: uuid::Uuid,
+    pub execution_job: Option<ExecutionJob>,
+    pub request_id: i64,
 }
 
 impl TryFrom<GetExecuteGeneratorResponseProto> for GetExecuteGeneratorResponse {
     type Error = PluginWorkQueueDeserializationError;
 
     fn try_from(value: GetExecuteGeneratorResponseProto) -> Result<Self, Self::Error> {
-        let request_id = value
-            .request_id
-            .ok_or(Self::Error::MissingRequiredField(
-                "GetExecuteGeneratorResponse.request_id",
-            ))?
-            .into();
-        let execution_job = value
-            .execution_job
+        let request_id = value.request_id;
+        let maybe_job = value.maybe_job.ok_or(Self::Error::MissingRequiredField(
+            "GetExecuteGeneratorResponseProto.maybe_job",
```

Comment:
Can tracing provide the context to know what proto we're dealing with, allowing us to just say `Error::MissingRequiredField("maybe_job")`?

Reply:
Honestly not sure. It depends on the context of the caller and how they handle the error (as well as all callers above them). We can't guarantee it, we could possibly expect it.